robottests/test_remotescript.robot

```diff
@@ -2,31 +2,249 @@
+    [Teardown]  Execute Command In Target  rm -rf /tmp/target_dir
+    Run Process     rm foo.sh  shell=${True}
+
+RemoteScript Create Directory In Target
+    Workaround Create Directory In Target
+    ${ret2}=    Execute Command In Target   stat /tmp/target_dir
+    Should Be Equal As Integers     ${ret2.status}  0   ${ret2}
+    [Teardown]  Execute Command In Target rm -r /tmp/target_dir
+
+RemoteScript Remove Directory In Target
+    Workaround Create Directory In Target
+    ${ret2}=    Remove Directory In Target  path=/tmp/target_dir
+    Should Be Equal As Integers     ${ret2.status}  0   ${ret2}
+    ${ret3}=    Execute Command In Target   stat /tmp/target_dir
+    Should Be Equal As Integers     ${ret3.status}  1   ${ret3}
+
+RemoteScript Set Target With Sshkeyfile
+    Set Target With Sshkeyfile  host=${HOST2.host}
+    ...                         username=${HOST2.user}
+    ...                         sshkeyfile=${HOST2.key}
+    ...                         name=host2
+    ${ret}=     Execute Command In Target   echo out    target=host2
+    Should Be Equal As Integers     ${ret.status}   0   ${ret}
+
+RemoteScript Get Target Properties
+    ${ret}=    Get Target Properties  target=default
+    Dictionary Should Contain Sub Dictionary     ${ret}   ${DEFAULT_TARGET}
+
+RemoteScript Execute Script In Target
+    Workaround Create Directory In Target
+    ${out}=     Run Process  ls ${CURDIR}  shell=${True}
+    Log  ${out.stdout}
+    ${ret2}=    Copy File To Target  source_file=${CURDIR}/target_script.sh  destination_dir=/tmp/target_dir
+    Should Be Equal As Integers     ${ret2.status}  0
+    ${ret3}=     Execute Script In Target    file=/tmp/target_dir/target_script.sh
+    Should Be Equal As Integers     ${ret3.status}  0
+    [Teardown]  Execute Command In Target  rm -r /tmp/target_dir
+
+RemoteScript Execute Background Command In Target
+    Execute Background Command In Target    command=echo Hello;echo World!>&2  target=default  exec_id=bg1
+    Execute Background Command In Target    command=echo Hello;echo World!>&2  target=default  exec_id=bg2
+    Execute Background Command In Target    command=echo Hello;echo World!>&2  target=default  exec_id=bg3
+    ${result_1}=    Wait Background Execution  bg1
+    ${result_2}=    Wait Background Execution  bg2
+    ${result_3}=    Wait Background Execution  bg3
+    Should Be Equal As Integers     ${result_1.status}     0
+    Should Be Equal As Integers     ${result_2.status}     0
+    Should Be Equal As Integers     ${result_3.status}     0
+    Should Be Equal      ${result_1.stdout}   Hello
+    Should Be Equal      ${result_2.stdout}   Hello
+    Should Be Equal      ${result_3.stdout}   Hello
+    Should Be Equal      ${result_1.stderr}   World!
+    Should Be Equal      ${result_2.stderr}   World!
+    Should Be Equal      ${result_3.stderr}   World!
+
+RemoteScript Execute Background Script In Target
+    Workaround Create Directory In Target
```

Comment:
Please change 'Workaround' to 'RemoteScript' in order to be consistent. The workaround can be added to the keyword (with TODO comment) as it is now.

Reply:
Changed Workaround to Better to avoid double naming issue.